modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kink_collars.dm

```diff
@@ -31,25 +31,32 @@
 	desc = "A nice, tight collar. It fits perfectly to your skin"
 	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_clothing/lewd_neck.dmi'
 	worn_icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_neck.dmi'
-	icon_state = "collar_cyan"
-	inhand_icon_state = "collar_cyan"
+	icon_state = "collar"
+	inhand_icon_state = "collar"
 	body_parts_covered = NECK
 	slot_flags = ITEM_SLOT_NECK
 	w_class = WEIGHT_CLASS_SMALL
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/small/kink_collar
 	var/tagname = null
 	var/treat_path = /obj/item/food/cookie
-	unique_reskin = list("Cyan" = "collar_cyan",
-						"Yellow" = "collar_yellow",
-						"Green" = "collar_green",
-						"Red" = "collar_red",
-						"Latex" = "collar_latex",
-						"Orange" = "collar_orange",
-						"White" = "collar_white",
-						"Purple" = "collar_purple",
-						"Black" = "collar_black",
-						"Black-teal" = "collar_tealblack",
-						"Spike" = "collar_spike")
+	var/current_color = "cyan"
+	var/color_changed = FALSE
+	var/static/list/collar_designs
+
+//create radial menu
+/obj/item/clothing/neck/kink_collar/proc/populate_collar_designs()
+	collar_designs = list(
+		"cyan" = image (icon = src.icon, icon_state = "collar_cyan"),
+		"yellow" = image(icon = src.icon, icon_state = "collar_yellow"),
+		"green" = image(icon = src.icon, icon_state = "collar_green"),
+		"red" = image(icon = src.icon, icon_state = "collar_red"),
+		"latex" = image(icon = src.icon, icon_state = "collar_latex"),
+		"orange" = image(icon = src.icon, icon_state = "collar_orange"),
+		"white" = image(icon = src.icon, icon_state = "collar_white"),
+		"purple" = image(icon = src.icon, icon_state = "collar_purple"),
+		"black" = image(icon = src.icon, icon_state = "collar_black"),
+		"tealblack" = image(icon = src.icon, icon_state = "collar_tealblack"),
+		"spike" = image(icon = src.icon, icon_state = "collar_spike"))
```

Comment:
can't you just append the name to the end of the icon state?

Reply:
(Hint: you can)